apps/src/Sounds.js

```diff
@@ -235,6 +239,52 @@ Sounds.prototype.checkDidSourcePlay_ = function(source, context, onComplete) {
   );
 };
 
+/**
+ * Registers the bytes of text to speech.
+ * @param {string} language the language the text is spoken in.
+ * @param {string} text the text that is being spoken.
+ * @param {boolean} hasProfanity
+ * @param {Array.<string>} profaneWords an array of profane words in the text if there are any. Otherwise null.
+ * @param {string} gender the gender of the voice used to speak the text. Optional
+ * @param {Array.<bytes>} bytes the bytes of the speech.
+ */
+Sounds.prototype.registerTextBytes = function(
+  language,
+  text,
+  hasProfanity,
+  profaneWords = null,
+  gender = null,
+  bytes = null
+) {
+  if (!this.textBytesByLanguage[language]) {
+    this.textBytesByLanguage[language] = {};
+  }
+  if (!this.textBytesByLanguage[language][text]) {
+    this.textBytesByLanguage[language][text] = {};
+  }
+  this.textBytesByLanguage[language][text]['hasProfanity'] = hasProfanity;
+  this.textBytesByLanguage[language][text]['profaneWords'] = profaneWords;
+  if (gender !== null) {
+    this.textBytesByLanguage[language][text][gender] = bytes;
+  }
+};
+
+/**
+ * Retrieves the bytes of text to speech if it exists.
+ * @param {string} language the language the text is spoken in.
+ * @param {string} text the text that is being spoken.
+ * @returns {Object} if text bytes found, null otherwise.
+ */
+Sounds.prototype.getTextBytes = function(language, text) {
+  if (
+    !this.textBytesByLanguage[language] ||
+    !this.textBytesByLanguage[language][text]
+  ) {
+    return null;
+  }
+  return this.textBytesByLanguage[language][text];
```

Comment:
nit: simplify to `return this.textBytesByLanguage.language && this.textBytesByLanguage.language.text`?